### PR TITLE
[20400] Test report action does not run when no-test label is set

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR prevents the Github actions windows CI from attempting to create a test report if the label `no-test` is set

@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- _N/A_: The code follows the style guidelines of this project. 
- _N/A_: Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_: Any new/modified methods have been properly documented using Doxygen. 
- _N/A_: Changes are ABI compatible. 
- _N/A_: Changes are API compatible. 
- _N/A_: New feature has been added to the `versions.md` file (if applicable).
- _N/A_: New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by EduPonz** [comment]
Cancelling rest of the CI to save resources

**Review by JesusPoderoso — DISMISSED** [review]
LGTM with green CI

File: .github/workflows/reusable-windows-ci.yml [hunk 1]
```diff
@@ -178,7 +178,7 @@ jobs:
 
       - name: Test Report
         uses: eProsima/eProsima-CI/external/test-reporter@main
-        if: success() || failure()
+        if: (success() || failure()) && ${{ !contains(github.event.pull_request.labels.*.name, 'no-test') }}
```
Comment (MiguelCompany) [inline .github/workflows/reusable-windows-ci.yml]:
```suggestion
        if: ${{ !cancelled() && !contains(github.event.pull_request.labels.*.name, 'no-test') }}
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI 😜

**Pull request merged**